CIP-0057/README.md

```diff
@@ -0,0 +1,414 @@
+---
+CIP: 57
+Title: Plutus Contract Blueprint
+Authors: KtorZ <matthias.benkort@cardanofoundation.org>, scarmuega <santiago@carmuega.me>
+Discussions-To: https://github.com/cardano-foundation/CIPs/pull/258
+Status: Draft 
+Type: Process
+Created: 2022-05-15
+License: CC-BY-4.0
+---
+
+## Abstract
+
+This document specifies a language for documenting Plutus contracts in a machine-readable manner. This is akin to what [OpenAPI](https://swagger.io/specification) or [AsyncAPI](https://www.asyncapi.com/docs/specifications/v2.4.0) are for, documenting HTTP services and asynchronous services respectively. In a similar fashion, A Plutus contract has a binary interface which is mostly defined by its datum and redeemer. Since all interactions with contracts are done through transactions, we can also model _endpoints_ of a contract as transitions from a known state to another purely declaratively. 
+
+This document is therefore a meta-specification defining the vocabulary and validation rules with which one can specify a Plutus contract interface, a.k.a **Plutus contract blueprint**.
+
+## Motivation
+
+While publicly accessible, on-chain contracts are currently inscrutable. Ideally, one would want to get an understanding of transactions revolving around script executions. This is both useful to visualize and to control the evolution of a contract life-cycle; but also, as a user interacting with a contract, to ensure that one is authorizing a transaction to do what it's intended to. Having a machine-readable specification in the form of a JSON-schema makes it easier (or even, possible) to enable a wide variety of use cases from a single concise document, such as:
+
+- Code generators for serialization/deserialization of Contract's elements
+- Contract API Reference / Documentation, also automatically generated
+- Extra automated transaction validation layers 
+- Better wallet UI / integration with DApps
+- Automated Plutus-code scaffolding 
+
+Moreover, by making the effort to write a clear specification of their contracts, DApps developers make their contracts easier to audit (as they're able to specify the expected behavior). 
+
+## Specification
+
+### Overview 
+
+This specification introduces the notion of a _Plutus contract blueprint_, as a JSON document which it itself a _JSON-schema_ as per the definition of given in [JSON Schema: A Media Type for Describing JSON Documents: Draft 2020-12](https://json-schema.org/draft/2020-12/json-schema-core.html).
```

Comment:
I don't think the encoding of the blueprint is important. You probably want to make this section much shorter and perhaps move it around.

Reply:
The encoding of the blueprint is the whole purpose of this CIP :)